horovod/_keras/__init__.py

```diff
@@ -79,13 +79,15 @@ def __init__(self, **kwargs):
                         scale_local_gradients=scale_local_gradients
                     )
 
-        def variables(self):
-            if _IS_TF2:
-                return super(self.__class__, self).variables()
-            return self.get_weights()
-
         if version.parse(tf.__version__) >= version.parse('2.12.0'):
-            variables = property(variables)
```

Comment:
Did this not work right in practice? (I only looked at unit tests..)

Reply:
I guess we didn't get the `opt.variables()` compatibility for the super class `variables()`..

Reply:
We had issues with RN50 tests at NVIDIA, getting `TypeError: 'list' object is not callable` failures, adding the shim that Keras repo uses fixed it.

Reply:
thanks for explaining the context 👍